src/main/java/com/google/cloud/genomics/utils/Contig.java

```diff
@@ -99,7 +99,7 @@ public Contig apply(String contigString) {
   // the ShardUtils methods should be used to ensure that shards are shuffled all together before
   // being returned to clients.
   List<Contig> getShards(long numberOfBasesPerShard) {
-    double shardCount = Math.ceil(end - start) / numberOfBasesPerShard;
+    double shardCount = (end - start) / (double) numberOfBasesPerShard;
```

Comment:
You are right that the double inside the calculation forces it to use the `Math.ceil(double ...)`, but you can still cast to `long` with `Math.ceil` and it works with the addition of `0.0`.  I just put this together based on the code in the PR, and it works with `Math.ceil` and `long`:

> Result:
> $ javac ShardTest.java ; java ShardTest
> true
> true
> true
> true
> true
> true
> true
> $

Below is the code:

```Java
import java.util.*;

public class ShardTest {
  
  public static void main( String [] args ) {

    Contig contig = new Contig("1", 0, 9);
    ArrayList<Contig> shards = contig.getShards(5);

    assertEquals(2, shards.size());
    Contig shard1 = shards.get(0);
    Contig shard2 = shards.get(1);

    assertEquals("1", shard1.referenceName);
    assertEquals(0, shard1.start);
    assertEquals(5, shard1.end);

    assertEquals("1", shard2.referenceName);
    assertEquals(5, shard2.start);
    assertEquals(9, shard2.end);
    
  }

  static void assertEquals( String a, String b ) {
    System.out.println( a.equals(b) );
  }

  static void assertEquals( long a, long b ) {
    System.out.println( a == b );
  }

  public static class Contig {

    public final String referenceName;
    public final long start;
    public final long end;

    public Contig(String referenceName, long start, long end) {
     this.referenceName = referenceName;
     this.start = start;
     this.end = end;
    }

    ArrayList<Contig> getShards(long numberOfBasesPerShard) {

      long shardCount = (long) Math.ceil( (end - start + 0.0) / numberOfBasesPerShard );
      ArrayList<Contig> shards = new ArrayList<Contig>();
      for (int i = 0; i < shardCount; i++) {
        long shardStart = start + (i * numberOfBasesPerShard);
        long shardEnd = Math.min(end, shardStart + numberOfBasesPerShard);

        shards.add(new Contig(referenceName, shardStart, shardEnd));
      }
      return shards;
    }
  }
}
```Java